From 22d4e5b52a13d1942bfbbad043e8046ea2aaa92c Mon Sep 17 00:00:00 2001
From: aobaharuki2005 <tranbaohnth@outlook.com>
Date: Thu, 28 May 2026 05:22:18 +0700
Subject: [PATCH 12/13] Fix: wrong #[allow(rustc::default_hash_types)] place

---
 cc-rs/src/tool.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cc-rs/src/tool.rs b/cc-rs/src/tool.rs
index 21e02cf4f01..35328e7a2c9 100644
--- a/cc-rs/src/tool.rs
+++ b/cc-rs/src/tool.rs
@@ -25,7 +25,7 @@
 /// compiler itself.
 #[derive(Clone, Debug)]
 #[allow(missing_docs)]
-#[allow(rustc::default_hash_types)]
+
 pub struct Tool {
     pub(crate) path: PathBuf,
     pub(crate) cc_wrapper_path: Option<PathBuf>,
@@ -38,6 +38,7 @@ pub struct Tool {
     pub(crate) has_internal_target_arg: bool,
 }
 
+#[allow(rustc::default_hash_types)]
 impl Tool {
     pub(crate) fn new(
         path: PathBuf,
-- 
2.50.1 (Apple Git-155)

